### PR TITLE
Fix error and add example with scripts. #1192

### LINKED ISSUE #1192: MovingAverage model yields wrong results
**Describe the bug**
See title

**Expected behavior**
If all values are above 0, the moving average should not fall below 0.

**Screenshots**
undefined

### PR DESCRIPTION
**Pull request by FWuellhorst**
Closes #1192

### CODE REVIEW

File: AixLib/Utilities/Math/Examples/MovingAverage.mo [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  Modelica.Blocks.Interfaces.RealOutput y "Continuous output signal"
+    annotation (Placement(transformation(extent={{100,-10},{120,10}})));
+equation
+
+  connect(movingAverage.y, y)
+    annotation (Line(points={{39.2,0},{110,0}}, color={0,0,127}));
+  connect(pulse.y, movingAverage.u)
+    annotation (Line(points={{-73,0},{-34.4,0}}, color={0,0,127}));
+
+annotation(experiment(Tolerance=1e-6, StopTime=50.0),
```
Comment (PMehrfeld) [inline AixLib/Utilities/Math/Examples/MovingAverage.mo:19]: I recommend to also add the solver here.

**Review by PMehrfeld — APPROVED** [review]

File: AixLib/Utilities/Math/Examples/MovingAverage.mo (hunk 1 above)
Comment (FWuellhorst) [reply AixLib/Utilities/Math/Examples/MovingAverage.mo:19]: I tried, but get the warning 
` Failed to interpret experiment annotation: solver = "CVode"`
or with method 
` Failed to interpret experiment annotation: method = "CVode"`
and could not find an example where this is specified. 
I specified method in the simulate and plot script however.

Comment (PMehrfeld) [reply AixLib/Utilities/Math/Examples/MovingAverage.mo:19]: You must add `Algorithm = "CVode"`.  
If you go into the simulation setup of Dymola and "store settings in model" you can check the keyword. Then it is `__Dymola_Algorithm` but you can change it to `Algorithm`.

**Review by PMehrfeld — APPROVED** [review]

File: bin/ci_whitelist/reference_check_whitelist.txt [hunk 2]
```diff
@@ -6,6 +6,4 @@ AixLib.Types
 
 AixLib.Utilities
 
-AixLib.Systems
-
```
Comment (DaJansenGit) [inline bin/ci_whitelist/reference_check_whitelist.txt:10]: how does it come that AixLib.Systems is deleted from whitelist with your changes?

**Review by DaJansenGit — DISMISSED** [review]
approved, but please check the annotation regarding `AixLib.Systems`

File: bin/ci_whitelist/reference_check_whitelist.txt (hunk 2 above)
Comment (FWuellhorst) [reply bin/ci_whitelist/reference_check_whitelist.txt:10]: Why would it be on there anyway? Sven told me to :D

**Comment by FWuellhorst** [comment]
@DaJansenGit and @PMehrfeld (If you are still interested ;))
The example `AixLib.ThermalZones.HighOrder.Examples.RoomGFOw2_DayNightMode`  got a numerical jacobian due to the combination of moving average and PI-Control of the thermostat. I fixed this by changing the moving average to a first order model. I think for temperature sensors, this modelling is more state of the art. Regarding time constant, I compared the trajectory to the old model. 
It's not the same, but the initial overshoot is damped in both cases.
Old moving average: 
grafik
New first order:
grafik

Do you agree with the changes?

File: AixLib/Utilities/Math/MovingAverage.mo [hunk 3]
```diff
@@ -5,13 +5,16 @@ model MovingAverage
...
+    final initType=Modelica.Blocks.Types.Init.InitialOutput,
+    final k=1,
+    final use_reset=false,
+    final y_start=u_start);
+  parameter Real u_start = 0 "Start value of input";
 
 equation
-
-  der(y)*aveTime = u - delay(u, aveTime);
+  y * aveTime = uInt.y - delay(uInt.y, aveTime);
```
Comment (DaJansenGit) [inline AixLib/Utilities/Math/MovingAverage.mo:16]: Checked and results are the same for the new moving average. But don't get yet why you restructured it after you are not using it in the thermostatic valve at all after your changes.
CPU Time is slower for the new variant so please explain your changes.

File: bin/CITests/02_UnitTests/CheckPackages/validatetest.py [hunk 4]
```diff
@@ -165,7 +165,7 @@ def _get_ch_models(self):
         lines = changed_models.readlines()
         for line in lines:
             if line.rfind(".mo") > -1 and line.find("package") == -1:
-                if line.find(self.package) > -1 and line.find("ReferenceResults") == -1:
+                if line.find(f'{self.mo_library}{os.sep}{self.package}') > -1 and line.find("ReferenceResults") == -1:
```
Comment (DaJansenGit) [inline bin/CITests/02_UnitTests/CheckPackages/validatetest.py:168]: where do these changes come from? is it only outdated and needs a dev remerge?

File: bin/CITests/05_Converter/google_charts.py [hunk 5]
```diff
@@ -224,6 +224,17 @@ def _createFolder(self, directory):
...
+        log_file = open(self.f_log, "r")
+        lines = log_file.readlines()
+        model_list = []
+        for line in lines:
+            if line.find("*** Warning:") > -1:
+                if line.find("*** Warning: Numerical Jacobian in 'RunScript") > -1 and line.find(".mos") > -1:
+                    model = line[line.rfind((os.sep)) :line.find(".mos")].lstrip()
+                    model_list.append(model)
+        return model_list
+
```
Comment (DaJansenGit) [inline bin/CITests/05_Converter/google_charts.py:237]: same here

File: bin/ci_whitelist/reference_check_whitelist.txt (hunk 2 above)
Comment (DaJansenGit) [reply bin/ci_whitelist/reference_check_whitelist.txt:10]: this is fine. Will be added by another merge as well.

File: AixLib/Utilities/Math/MovingAverage.mo (hunk 3 above)
Comment (FWuellhorst) [reply AixLib/Utilities/Math/MovingAverage.mo:16]: As discussed

**Pull request merged**